libphal/phal_dump.C

```diff
@@ -99,37 +104,82 @@ struct DumpPPERegValue {
 	}
 } __attribute__((packed));
 
+// Define the chip types numbers
+static constexpr int PROC_SBE_DUMP = 0xA;
+static constexpr int ODYSSEY_SBE_DUMP = 0xB;
+
 /**
- * @brief Return the proc target corresponding to failing unit
- * @param[in] failingUnit Position of the proc containing the failed SBE
+ * @brief Get the Fapi Unit Pos object for OCMB
  *
- * @return Pointer to the pdbg target for proc
+ * @param target The target
+ * @return uint8_t The position
+ */
+uint32_t getFapiUnitPos(pdbg_target* target)
+{
+	uint32_t fapiUnitPos = 0; // chip unit position
+	if (!pdbg_target_get_attribute(target, "ATTR_FAPI_POS", 4, 1,
+				       &fapiUnitPos))
+		log(level::ERROR, "ATTR_FAPI_POS Attribute get failed");
+
+	return fapiUnitPos;
+}
+
+/**
+ * @brief Return the chip target corresponding to failing unit
+ * @param[in] failingUnit Position of the ocmb containing the failed SBE
+ * @param sbeTypeId The chip type number
+ *
+ * @return Pointer to the pdbg target for retreived chip
  *
  * Exceptions: PDBG_TARGET_NOT_OPERATIONAL if target is not available
  */
-struct pdbg_target* getProcFromFailingId(const uint32_t failingUnit)
+struct pdbg_target* getTargetFromFailingId(const uint32_t failingUnit,
+					   const int sbeTypeId)
 {
-	struct pdbg_target* proc = NULL;
-	struct pdbg_target* target;
-	pdbg_for_each_class_target("proc", target)
+	struct pdbg_target* chip = nullptr;
+	struct pdbg_target* target = nullptr;
+	std::string chipTypeString;
+
+	if (sbeTypeId == ODYSSEY_SBE_DUMP)
+		chipTypeString = "ocmb";
+	else if (sbeTypeId == PROC_SBE_DUMP)
+		chipTypeString = "proc";
+
+	pdbg_for_each_class_target(chipTypeString.c_str(), target)
 	{
-		if (pdbg_target_index(target) != failingUnit) {
+		uint8_t targetIdx = 0;
+		if (sbeTypeId == ODYSSEY_SBE_DUMP)
+			targetIdx = getFapiUnitPos(target);
```

Comment:
is FAPI unit position is a unique number across the system?

Reply:
Yes. Earlier without moving ocmb out, the index positions were repeated.. But Fapi positions are unique. As ocmb is not yet moved out, this is the best thing we can do. Confirmed by @deepakala-k